src/fromager/vendor_rust.py

```diff
@@ -119,11 +118,42 @@ def vendor_rust(
     ``setuptools-rust`` or ``maturin``, and has a ``Cargo.toml``, otherwise
     ``False``.
     """
-    if not _should_vendor_rust(req, project_dir):
+    pyproject_toml = dependencies.get_pyproject_contents(project_dir)
+    if not pyproject_toml:
+        logger.debug(f"{req.name}: has no pyproject.toml")
         return False
 
-    # check for Cargo.toml
-    manifests = list(project_dir.glob("**/Cargo.toml"))
+    backend = _detect_rust_build_backend(req, pyproject_toml)
+    if backend is None:
+        return False
+
+    manifests: list[pathlib.Path] = []
+    # check for Cargo.toml in project root
+    root_cargo_toml = project_dir / "Cargo.toml"
+    if root_cargo_toml.is_file():
+        manifests.append(root_cargo_toml)
+
+    # maturin and setuptools-rust can have Cargo.toml in other directories.
+    # tool.setuptools-rust and tool.maturin are optional.
+    if backend == "maturin":
```

Comment:
This if/elif block could use a final `else` to report that an unknown backend is being used so no additional cargo files are being scanned. I've found a few places in our code where we didn't emit a message like that and ended up with confusing behavior.

Reply:
The if/elif block is now a match block that handles all cases (maturin, setuptools-rust, None, unreachable). I'm also printing a warning when we detect a Rust build backend but no Cargo.toml.